### DISCUSSION TITLE
Detailed instructions for two step MBR imitation in a pipeline

### DISCUSSION THREAD

**Discussion by sooheon** [vdemichev/DiaNN#1139] [Q&A] [answered]
This is the documentation for manually generating predicted speclib from FASTA + DIA then running analysis with improved speclib (i.e. match-between-runs): 

> One can manually 'imitate' MBR using a two-step approach that will result in comparable performance. First, run DIA-NN to create a spectral library from the DIA runs (the whole experiment or just its subset, which can be a lot quicker for large-scale experiments or experiments including blanks/failed runs). Then use this library to analyse the whole experiment. In either case, run DIA-NN with MBR disabled.

Please let me know if this is correct. 

Step 1:
- Select subset of raw files for FASTA + DIA speclib generation
- Select "FASTA digest for library-free search / library generation"
  - Some checkboxes are selected by default: "Deep learning-based spectra, RTs and IMs prediction", "Generate spectral library", "MBR". Turn _off_ "MBR".
- Select modifications / peptide / precursor settings as appropriate 

Step 2:
- Use all raw files of experiment
- Select generated speclib from step 1 as "Spectral library" input
- Do I uncheck "FASTA digest" and "Deep learning-based spectra" or keep it the same?
  - Do I set the same precursor ion generation settings (peptide length, precursor charge...) or are these only used for the initial speclib generation, so irrelevant?
- Keep "MBR" off
- Do I uncheck "Generate spectral library"?

Using these two steps as a pipeline, how do I set speclib input for Step 2 if it hasn't been generated yet? Do I just create an empty file with the same name as output speclib and assume it will be overwritten?

**Comment by vdemichev** [comment, marked as answer]
Hi Sooheon,

Step 0: generate a predicted spectral library (no raw files specified during this). This also applies to a normal MBR search, should start with a predicted (or empirical) library. 
Step 1: In this case, 'FASTA digest' and 'Deep learning' should be unchecked, as the library already contains predicted spectra, RTs and IMs. FASTA still needs to be provided, but purely for protein annotation. Modifications should be selected if you want them scored, please see the PTMs section in the docs.
Step 3: 'FASTA digest' and 'Deep learning' should be unchecked. Precursor generation settings: only modifications specified matter, everything else has no effect.

> Do I uncheck "Generate spectral library"?

Yes.

> how do I set speclib input for Step 2 if it hasn't been generated yet?

The input lib is in .tsv format (1.9) or .parquet (1.9.1), you can just type the right file name in 'Spectral library' field, even if the file does not exist yet.

Best,
Vadim

**Reply by sooheon** [reply]
Thank you. 

> Step 0:

If I just run Step 1 and provide FASTA, does this do Step 0 and 1 together?

Is it ok to leave all precursor ion generation settings default for Step 0, and only add PTM settings at step 1 and 2?

> Precursor generation settings: only modifications specified matter, everything else has no effect.

Do you mean the checkbox only? (i.e. if I check Phospho but have different "Number of variable modifications" this is OK?

**Reply by vdemichev** [reply]
> If I just run Step 1 and provide FASTA, does this do Step 0 and 1 together?

Yes, but this is not recommended, and in versions before 1.9.1 the results are not identical to doing the steps separately.

> Is it ok to leave all precursor ion generation settings default for Step 0, and only add PTM settings at step 1 and 2?

No, if you want modifications added, they naturally must be present in the library, otherwise they will not be searched.

> Do you mean the checkbox only?

Yes, number of var mods will not have an effect on search results, only lib generation.
